openvdb_cmd/vdb_tool/include/Geometry.h

```diff
@@ -393,8 +407,16 @@ void Geometry::read(const std::string &fileName)
         this->readGEO(fileName);
         break;
     default:
-      throw std::invalid_argument("Geometry::read: File \""+fileName+"\" has an invalid extension");
-      break;
+#if VDB_TOOL_USE_PDAL
```

Comment:
this looks strange to me. Why is this file type handles in the default case and not according to a file extension like the other file types?

Reply:
Indeed, you've never come across a pdal file because pdal is not a file type itself, it is the "point data abstraction library"  pdal.io

PDAL supports many different file types through its own plugins - such as LAS, E57, PLY and so forth.  
https://pdal.io/en/2.8.1/stages/readers.html

As it is pluggable, I couldn't simply code in a supported file extension or list of extensions, and used `pdal::StageFactory::inferReaderDriver(fileName)` to determine if the file can be opened.  This was as neat as I could be without interrogating the library to build up a list of supported file extensions - which it doesn't directly let me access, they are referred to as `drivers` and don't need to match the extension.  

We could add a test for a .las or .e57 or even .npy file as they are supported by default, but it might be overkill to test every supported type.

Reply:
Hi @kmuseth, per our discussion I added a unittest which will use PDAL to read an ASCII formatted file that it supports.

Reply:
now that I understand that pdal can import multiple file types I do agree that it makes sense to place this is the "default" case. So, please ignore my comment above. The only remaining thing a minimum unit-test to this code at get's compiled and run during the CI

Reply:
Great, I've added that unit test which uses PDAL functionality to read some points similar to other similar format readers' unit tests.